### DISCUSSION TITLE
Otel Collector deployment on Kubernetes

### DISCUSSION THREAD

**Discussion by dimitri995** [open-telemetry/opentelemetry-collector#6362] [Q&A]
Hi everyone !

I need an architectural advice on the deployment of the Collector on Kubernetes. The goal is to trace kafka message send/consume by externals applications using our lib (framework SpringKafka). Traces are sent by our lib with OpenTelemetry. But I don't know where I need to have my collector.

In sedond time we need to trace every query in other pods.. 

I have seen 2 mains architecture : 
> 1.  As a DaemonSet with a collector pod on every single cluster node
![image](https://user-images.githubusercontent.com/66120155/196915193-9cf498b4-0daf-47bf-a12c-10e55715df33.png)

> 2.  As a sidecars to the application pods
![image](https://user-images.githubusercontent.com/66120155/196915301-348dd5e5-24c2-4cd3-a23c-6354e76f6df1.png)

I would want to know if there is a possibility and if it's a good practice to have our "Admin cluster" with all we need to Monitor (Collector, Jaeger, Prometheus, Grafana...)

![KubeTracing archi(1)](https://user-images.githubusercontent.com/66120155/196921710-ad3e4265-7391-4850-abb6-7677ab03668b.png)

Thanks for your time,

Regards,

Dimitri

**Comment by jpkrohling** [comment]
Absolutely, what you called "admin cluster" is what I called "control plane" or "monitoring" cluster here:

https://www.youtube.com/watch?v=WhRrwSHDBFs

and here: https://github.com/jpkrohling/opentelemetry-collector-deployment-patterns/tree/main/pattern-3-kubernetes
